ethereumj-core/src/main/java/org/ethereum/core/BlockchainImpl.java

```diff
@@ -867,7 +876,8 @@ private BlockSummary applyBlock(Repository track, Block block) {
             stateLogger.debug("apply block: [{}] tx: [{}] ", block.getNumber(), i);
 
             Repository txTrack = track.startTracking();
-            TransactionExecutor executor = new TransactionExecutor(tx, block.getCoinbase(),
+            TransactionExecutorFactory txFactory = commonConfig.transactionExecutorFactory();
```

Comment:
Inject this factory with Spring?

Reply:
It'll be a bit simpler in using, but increases difficulty for testing. I will try to find good trade-off in such cases.

Reply:
ok, maybe we will just get rid of that factory